elements/pl-hidden-hints/pl-hidden-hints.py

```diff
@@ -0,0 +1,77 @@
+import chevron
+import lxml
+import prairielearn as pl
+from typing import TypedDict, Tuple, List
+
+
+class HintsDict(TypedDict):
+    """A class with type signatures for the partial scores dict"""
+    html_hints: List[Tuple[int, str]]
+
+
+def get_param_key(name_prefix: str) -> str:
+    return f'{name_prefix}_hidden_hints'
+
+
+def prepare(element_html: str, data: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+    pl.check_attribs(element, [], ['name'])
+
+    name_prefix = pl.get_string_attrib(element, 'name', '')
+    param_key = get_param_key(name_prefix)
+
+    # Set param dict using prefix. The use of a prefix allows for multiple hint panels.
+    if (param_key in data['params']):
+        raise ValueError(f"Duplicate params key: '{param_key}'.")
+
+    hints_dict = data['params'].setdefault(param_key, dict())
+
+    # Parse hints from frontend
+    html_hints = []
+
+    # Read hints defined as child tags and load them into html_hints
+    for child in element:
+        if child.tag == 'pl-hint':
+            pl.check_attribs(child, [], ['show-after'])
+
+            # Default show-after to -1 to automatically show hint (closed) at start
+            priority = pl.get_integer_attrib(child, 'show-after', -1)
+            html_hints.append((priority, pl.inner_html(child)))
+        elif child.tag is lxml.etree.Comment:
+            continue
+        else:
+            raise Exception(f"Tags inside of pl-hidden-hints must be pl-hint, not '{child.tag}'.")
+
+    hints_dict['html_hints'] = html_hints
```

Comment:
Is it necessary to store anything in `params` at all? I'd almost expect this element to be completely stateless - that is, it looks like this element will always render deterministically based on just the question HTML and `num_valid_submissions`.

The advantage to always treating the HTML as the source of truth is that if a hint is adjusted by an instructor, students will be able to see it the next time they reload the page; they won't have to wait until they generate a new variant (if they can at all).

Reply:
I think doing this made more sense for our internal version, but you're right that it will cause less confusion if we recompute this on the fly (using HTML as groundtruth). Will make those changes.